cmd/object-handlers.go

```diff
@@ -806,28 +837,19 @@ func (api objectAPIHandlers) CompleteMultipartUploadHandler(w http.ResponseWrite
 		return
 	}
 
-	doneCh := make(chan struct{})
-
-	// Signal that completeMultipartUpload is over via doneCh
+	// Signal that joining parts is over is over via doneCh
 	go func(doneCh chan<- struct{}) {
-		md5Sum, err = objectAPI.CompleteMultipartUpload(bucket, object, uploadID, completeParts)
+		md5Sum, err = joinParts()
 		doneCh <- struct{}{}
 	}(doneCh)
 
 	sendWhiteSpaceChars(w, doneCh)
```

Comment:
Let's remove this and we will solve all the problems and we will not need to change the function signature of `CompleteMultipartUpload`